internal/bundledeployment/bundledeployment.go

```diff
@@ -0,0 +1,244 @@
+/*
```

Comment:
Commenting here to call specific attention to this file. How much more of the k8s stuff here can be removed? Do we need the `json:"..."` tags? Are we OK with the fact that I copied an autogenerated func into this file in order to maintain `DeepCopy()`?